### PR TITLE
Rewrite  the mac syslog tests as they seem somewhat flaky.

### PR DESCRIPTION
**Pull request by JamesWTruher**
# PR Summary

Change the way logging tests are done on mac, these tests often fail, especially in our release cycle testing.

## PR Context

the current method for collecting syslog entries on mac uses the _text_ output from MacOS `log` and then using a lot of text parsing which can be a bit fragile. However, there is an option for outputting json which reduces the amount of parsing greatly. With this PR we're now only parsing the EventMessage for the EventId which is far more regular. This PR also reduces the complexity in just launching the `log` app. Hopefully, this will reduce the number of failures of these tests. I was also able to re-enable a pending test.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: test/powershell/Host/Logging.Tests.ps1 [hunk 1]
```diff
@@ -317,26 +340,19 @@ Path:.*
...
 
-    It 'Verifies basic logging with no customizations' -Skip:(!$IsSupportedEnvironment) {
+    It 'Verifies basic logging with no customizations' -Skip:(!$IsMacOS) {
         try {
+            $timeString = [DateTime]::Now.ToString('yyyy-MM-dd HH:mm:ss')
             $configFile = WriteLogSettings -LogId $logId
+            copy-item $configFile /tmp/pwshtest.config.json
             $testPid = & $powershell -NoProfile -SettingsFile $configFile -Command '$PID'
-
-            Export-PSOsLog -After $after -LogPid $testPid -TimeoutInMilliseconds 30000 -IntervalInMilliseconds 3000 -MinimumCount 3 |
```
Comment (daxian-dbw) [inline test/powershell/Host/Logging.Tests.ps1:325]: `Export-PSSysLog` is no longer used, so maybe we should remove it from the `PSSysLog` module?

File: test/powershell/Host/Logging.Tests.ps1 [hunk 2]
```diff
@@ -43,6 +43,29 @@ enum LogKeyword
     ManagedPlugin = 0x100
 }
 
+# mac log command can emit json, so just use that
+# we need to deconstruct the eventmessage to get the event id
+# we also need to filter out the non-default messages
+function Get-MacOsSyslogItems {
```
Comment (daxian-dbw) [inline test/powershell/Host/Logging.Tests.ps1:49]: Given that this is to replace `Get-PSOsLog`, does it make sense to put this function in `PSSysLog.psm1`?

**Review by daxian-dbw — COMMENTED** [review]
Looks good to me. Left 2 minor comments.

File: test/powershell/Host/Logging.Tests.ps1 (hunk 2 above)
Comment (JamesWTruher) [reply test/powershell/Host/Logging.Tests.ps1:49]: @daxian-dbw I did think about that, but wanted to take more time to do a more complete rewrite of this module. I thought this smaller, localized change would be a faster path to stability, so I took this parallel approach.

Comment (daxian-dbw) [reply test/powershell/Host/Logging.Tests.ps1:49]: Sounds good to me 👍

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**